tests/integration/test_fim/conftest.py

```diff
@@ -55,8 +73,63 @@ def install_audit(get_configuration):
         audit = "auditd"
         option = "--yes"
     else:
-        raise ValueError(f"Linux distro ({linux_distro}) not supported for uninstall/install audit")
+        # Install audit and start the service
+        process = subprocess.run([package_management, "install", audit, option], check=True)
+        process = subprocess.run(["service", "auditd", "start"], check=True)
+
+
+@pytest.fixture(scope='session')
+def configure_local_internal_options_fim():
+    """Fixture to configure the local internal options file."""
+
+    # Backup the old local internal options
+    backup_local_internal_options = get_wazuh_local_internal_options()
+
+    # Set the new local internal options configuration
+    set_wazuh_local_internal_options(create_local_internal_options(FIM_DEFAULT_LOCAL_INTERNAL_OPTIONS))
+
+    yield
+
+    # Backup the old local internal options cofiguration
+    set_wazuh_local_internal_options(backup_local_internal_options)
+
 
-    # Install audit and start the service
-    process = subprocess.run([package_management, "install", audit, option], check=True)
-    process = subprocess.run(["service", "auditd", "start"], check=True)
+@pytest.fixture(scope='function')
+def set_wazuh_configuration_fim(configuration, set_wazuh_configuration, configure_local_internal_options_fim):
+    """Set wazuh configuration
+
+    Args:
+        configuration (dict): Configuration template data to write in the ossec.conf.
+        set_wazuh_configuration (fixture): Set the wazuh configuration according to the configuration data.
+        configure_local_internal_options_fim (fixture): Set the local_internal_options.conf file.
+    """
+    yield
+
+
+@pytest.fixture(scope='function')
+def wait_for_fim_start_function(configuration):
+    """
+    Wait for realtime start, whodata start or end of initial FIM scan.
+    """
+    file_monitor = FileMonitor(LOG_FILE_PATH)
+    mode_key = 'fim_mode' if 'fim_mode2' not in configuration else 'fim_mode2'
+
+    try:
+        if configuration[mode_key] == 'realtime':
+            detect_realtime_start(file_monitor)
+        elif configuration[mode_key] == 'whodata':
+            detect_whodata_start(file_monitor)
+        else:  # scheduled
+            detect_initial_scan(file_monitor)
+    except KeyError:
+        detect_initial_scan(file_monitor)
+
+
+@pytest.fixture(scope="function")
+def restart_syscheck_function():
+    """
+    Restart syscheckd daemon.
+    """
+    control_service("stop", daemon="wazuh-syscheckd")
+    truncate_file(LOG_FILE_PATH)
+    control_service("start", daemon="wazuh-syscheckd")
```

Comment:
Why do you add it there? In the `integration/conftest.py` file has defined it.

When you create a new method, you should make sure it doesn't exist elsewhere so as not to duplicate code.

Reply:
This version This has been added here, following the structure applied to the VDT test suite refactor. Having a specific function that restarts only syscheck allows for tests to be a bit cleaner by not needing to pass the daemon to stop as a parameter.